### PR TITLE
Add vote federation change unauthorized

### PR DESCRIPTION
**Pull request by jeremy-then**
Add vote federation change unauthorized

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 1]
```diff
@@ -2205,4 +2244,11 @@ private List<ECKey> getMstPublicKeysFromFederationMembers(List<FederationMember>
             .map(FederationMember::getMstPublicKey)
             .collect(Collectors.toList());
     }
+
+    private Transaction getTransactionFromCaller(RskAddress authorizer) {
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2248]: We have this in `TransactionUtils` class

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 2]
```diff
@@ -2194,6 +2200,39 @@ void save_callsStorageProviderSave() {
...
+            bridgeEventLogger = mock(BridgeEventLogger.class);
+        }
+
+        @Test
+        void voteFederationChange_WithUnauthorizedCaller_returnsUnauthorizedResponseCode() {
+
+            // Arrange
+
+            Transaction tx = getTransactionFromCaller(FederationChangeCaller.UNAUTHORIZED.getRskAddress());
+            ABICallSpec abiCallSpec = new ABICallSpec("create", new byte[][]{});
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 3]
```diff
@@ -2194,6 +2200,39 @@ void save_callsStorageProviderSave() {
...
+    @Tag("vote federation change")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class VoteFederationChangeTest {
+
+        BridgeEventLogger bridgeEventLogger;
+
+        @BeforeAll
+        void setUp() {
+            signatureCache = mock(SignatureCache.class);
+            bridgeEventLogger = mock(BridgeEventLogger.class);
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 4]
```diff
@@ -2194,6 +2202,47 @@ void save_callsStorageProviderSave() {
...
+            bridgeEventLogger = new BridgeEventLoggerImpl(BridgeMainNetConstants.getInstance(), activations, Collections.EMPTY_LIST, signatureCache);
+            federationSupport = federationSupportBuilder
+                .withFederationConstants(federationMainnetConstants)
+                .withFederationStorageProvider(storageProvider)
+                .withActivations(activations)
+                .build();
+        }
+
+        @Test
+        void voteFederationChange_WithUnauthorizedCaller_returnsUnauthorizedResponseCode() {
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:2227]: `_with`

**Review by julia-zack — APPROVED** [review]
lgtm, left minor comment

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**